### PR TITLE
Bugfix/20210605 xlabel

### PR DESCRIPTION
**Pull request by timtreis**

### CODE REVIEW

File: R/visr.R [hunk 1]
```diff
@@ -235,11 +235,18 @@ visr.survfit <- function(
 
 # Obtain X-asis label ----------------------------------------------------------
 
-  if (is.null(x_label)){
+  if (is.null(x_label)) {
+    
     if ("PARAM" %in% names(x)) x_label = x[["PARAM"]]
     if (! "PARAM" %in% names(x)) x_label = "time"
```
Comment (SHAESEN2) [inline R/visr.R:241]: Then x_label = what the user specifies. Don't force "time".

File: R/visr.R [hunk 2]
```diff
@@ -235,11 +235,18 @@ visr.survfit <- function(
...
     if (!is.null(x_units)) x_label = paste0(x_label, " (", x_units, ")")
```
Comment (SHAESEN2) [inline R/visr.R:242]: this does not make sense. if the label is not specified, then units should not be added as well. You agree looking at the testing script :)

File: tests/testthat/test-visr_plot.R [hunk 3]
```diff
@@ -332,6 +333,27 @@ testthat::test_that("T2.15 An error when an undefined option is passed to `legen
   
 })
 
+testthat::test_that("T2.16 When both `x_label` and `x_units` are specified, they are concatenated into the x-axis label.", {
```
Comment (SHAESEN2) [inline tests/testthat/test-visr_plot.R:336]: add tests for one of them not specified and a test for both not specified

**Review by SHAESEN2 — CHANGES_REQUESTED** [review]

File: R/visr.R (hunk 1 above)
Comment (timtreis) [reply R/visr.R:241]: Well, I'm not :D 

If `x_label` and `x_units` are both not `NULL`, they are concatenated to `<x_label> (<x_units>)`.

If `x_label` is not `NULL`, but `x_units` is it set's the label to `x_label`.

If `x_label` is `NULL`, but `x_units` is not, it tries to deduct the name from the data. If that is not possible, it's set to the default "time". And then it appends `x_units`.

If `x_label` and `x_units` are `NULL`, it tries to deduct the name from the data. If that is not possible, it's set to the default "time".

File: R/visr.R (hunk 2 above)
Comment (timtreis) [reply R/visr.R:242]: I think it does make sense, as for example in the example below.

```
adtte %>%
  visR::estimate_KM("SEX") %>%
  visR::visr(x_label = NULL,
             x_units = "days")
```
image

File: tests/testthat/test-visr_plot.R (hunk 3 above)
Comment (timtreis) [reply tests/testthat/test-visr_plot.R:336]: T2.1 and T2.3 cover this behaviour since their default is `NULL`.

File: R/visr.R (hunk 1 above)
Comment (SHAESEN2) [reply R/visr.R:241]: Dont use "time" as default. Default is NULL.

File: R/visr.R (hunk 2 above)
Comment (SHAESEN2) [reply R/visr.R:242]: Sure but here the label is extracted from the dataset. When no PARAM is available then it does not make sense.

**Comment by timtreis** [comment]
Changes that were to introduced here were now part of [PR 212](https://github.com/openpharma/visR/pull/212) which makes this PR unnecessary.

**Pull request closed without merging**